apps/studio/src/utils/sitemap.ts

```diff
@@ -65,30 +105,36 @@ const getSitemapTreeFromArray = (
     const indexPage = resources.find(
       (child) =>
         // NOTE: This child is the index page of this resource
-        child.permalink === INDEX_PAGE_PERMALINK &&
-        child.parentId === resource.id,
+        child.type === ResourceType.IndexPage && child.parentId === resource.id,
     )
 
-    const titleOfPage = indexPage?.title
-    const summaryOfPage = indexPage?.summary
+    if (!!indexPage) {
+      return {
+        id: String(resource.id),
+        layout: "content", // Note: We are not using the layout field in our previews
+        title: indexPage.title,
+        summary: indexPage.summary ?? `Pages in ${resource.title}`,
+        lastModified: new Date() // TODO: Update this to the updated_at field in DB
+          .toISOString(),
+        // NOTE: This permalink is unused in the preview
+        permalink: resource.permalink,
+        image: !!indexPage.thumbnail
+          ? { src: indexPage.thumbnail, alt: "" }
+          : undefined,
+        children: getSitemapTreeFromArray(resources, resource.id),
+      }
+    }
 
     return {
       id: String(resource.id),
       layout: "content", // Note: We are not using the layout field in our previews
-      title: titleOfPage || resource.title,
-      summary: summaryOfPage ?? `Pages in ${resource.title}`,
+      title: resource.title,
+      summary: `Pages in ${resource.title}`,
       lastModified: new Date() // TODO: Update this to the updated_at field in DB
         .toISOString(),
```

Comment:
nit: can update this also since the information should be in `resource` now anyway

Reply:
updated